packages/editor/src/store/effects/posts.js

```diff
@@ -81,9 +81,18 @@ export const requestPostUpdate = async ( action, store ) => {
 		edits = { status: 'draft', ...edits };
 	}
 
+	const content = getEditedPostContent( state );
+
+	// The edited post property is not kept in sync with blocks state. At save
+	// time, we derive the content and set the edit before calling the update,
+	// such that it is considered synced at the time of the optimistic update,
+	// and divergences in the persisted value are accounted for in the post
+	// reset which follows (i.e. mark as dirty if saved content differs).
+	dispatch( editPost( { content } ) );
```

Comment:
I'm not certain I follow why we need this?
Is it an optimistic update, if it's the case should we add the `optimist: { id: POST_UPDATE_TRANSACTION_ID }` ?

Reply:
It's probably not a good sign that my initial thought is I didn't quite remember why I added it. I'll see if I can improve the comment.

I think the issue was that without it, the content was being considered as an unsaved edit because we compare what we receive via `updatePost` against `edits`, keeping those which differ.